src/CFG.hs

```diff
@@ -381,9 +382,18 @@ go_staged supply context e =
 
       | otherwise -> error "impossible"
 
-elem_c :: (Lift c, Eq c) => Code c -> [c] -> Code Bool
-elem_c c [] = [|| False ||]
-elem_c c (x:xs) = [|| if x == $$(c) then True else $$(elem_c c xs) ||]
+elem_c
```

Comment:
Can you explain how the version of `elem` before produced worse code?

Reply:
I don't know what sort of optimisation is happening here, but here's a picture of some of the old code https://i.imgur.com/ACXijnG.png